tests/pytest/test_dragonfly.py

```diff
@@ -7,18 +7,26 @@
 
 SCRIPT_DIR = os.path.dirname(os.path.abspath(__file__))
 
-@pytest.fixture(scope="module")
-def df_server():
+DRAGONFLY_PATH = os.environ.get("DRAGONFLY_HOME", os.path.join(SCRIPT_DIR, '../../build-dbg/dragonfly'))
```

Comment:
Wow! This is super! 
I do not have enough python knowledge to actually review your code 😆
I have a question - does it start and stop a df process for each test or it can reuse the same process between multiple tests?  
(It's  not important right now, but once the tests grow I want to be mindful about their running time)

Reply:
Yes you can reuse the process between multiple tests. A process is made for each test class (and each parameter configuration defined by the class) but each test function within the object is run with the same process.